src/main/java/com/acrolinx/sidebar/utils/NamespaceResolver.java

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2023-present Acrolinx GmbH */
+package com.acrolinx.sidebar.utils;
+
+import java.util.Iterator;
+import javax.xml.XMLConstants;
+import javax.xml.namespace.NamespaceContext;
+import org.w3c.dom.Document;
+
+public class NamespaceResolver implements NamespaceContext {
+  private final Document sourceDocument;
+
+  public NamespaceResolver(Document document) {
+    sourceDocument = document;
+  }
+
+  @Override
+  public String getNamespaceURI(String prefix) {
+    if (prefix.equals(XMLConstants.DEFAULT_NS_PREFIX)) {
+      return sourceDocument.lookupNamespaceURI(null);
+    } else {
+      return sourceDocument.lookupNamespaceURI(prefix);
+    }
+  }
+
+  @Override
+  public String getPrefix(String namespaceURI) {
```

Comment:
According to https://google.github.io/styleguide/javaguide.html#s5.3-camel-case the parameter name should be "`namespaceUri`" (I would even suggest "`namespaceUriString`"). Also in the method below

Reply:
Not fixed yet

Reply:
> Not fixed yet

Done